src/libos/src/fs/inode_file.rs

```diff
@@ -182,32 +187,62 @@ impl File for INodeFile {
         Ok(())
     }
 
-    fn test_advisory_lock(&self, lock: &mut Flock) -> Result<()> {
-        // Let the advisory lock could be placed
-        // TODO: Implement the real advisory lock
-        lock.l_type = FlockType::F_UNLCK;
+    fn test_advisory_lock(&self, lock: &mut RangeLock) -> Result<()> {
+        if self.inode.ext().is_err() {
+            warn!("Inode extension is not supportted, the lock could be placed");
+            lock.set_type(RangeLockType::F_UNLCK);
+            return Ok(());
```

Comment:
Let's make the FS which not implements ext() with a dummy implementation that one can always place the lock. This is because If we return an error, some demos in CI will fail since our prior version is the dummy implementation.